components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/MyAccountAuthorizedAPIListener.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.mgt.listener;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.api.resource.mgt.APIResourceMgtException;
+import org.wso2.carbon.identity.api.resource.mgt.constant.APIResourceManagementConstants;
+import org.wso2.carbon.identity.api.resource.mgt.util.APIResourceManagementUtil;
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+import org.wso2.carbon.identity.application.common.model.APIResource;
+import org.wso2.carbon.identity.application.common.model.AuthorizedAPI;
+import org.wso2.carbon.identity.application.common.model.AuthorizedScopes;
+import org.wso2.carbon.identity.application.common.model.Scope;
+import org.wso2.carbon.identity.application.mgt.ApplicationConstants;
+import org.wso2.carbon.identity.application.mgt.ApplicationManagementService;
+import org.wso2.carbon.identity.application.mgt.internal.ApplicationManagementServiceComponentHolder;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * MyAccount authorized API listener.
+ */
+public class MyAccountAuthorizedAPIListener extends AbstractAuthorizedAPIManagementListener {
+
+    private static final List<String> authorizedNoPolicyAPIIdentifiers = Arrays.asList(
+            "/api/users/v1/me/approval-tasks",
+            "/o/api/users/v1/me/approval-tasks");
+    private static final List<String> authorizedNoPolicyScopes = Arrays.asList(
+            "internal_approval_task_view",
+            "internal_approval_task_update",
+            "internal_org_approval_task_view",
+            "internal_org_approval_task_update");
+
```

Comment:
We can add authorization required APIs and Scopes also.
May be in a different PR

Reply:
Yes at the moment, focused only on approval-tasks API, in future, if other APIs with RBAC is needed, this can be improved.